services/util/test/TestSerialCommunicationLoopback.cpp

```diff
@@ -0,0 +1,59 @@
+#include "infra/timer/test_helper/ClockFixture.hpp"
+#include "infra/util/ByteRange.hpp"
+#include "infra/util/Function.hpp"
+#include "infra/util/test_helper/MockCallback.hpp"
+#include "services/util/SerialCommunicationLoopback.hpp"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+class SerialCommunicationLoopbackTest
+    : public testing::Test
+    , public infra::ClockFixture
+{
+public:
+    services::SerialCommunicationLoopback SerialCommunicationLoopback;
+    services::SerialCommunicationLoopbackPeer& server{ SerialCommunicationLoopback.Server() };
+    services::SerialCommunicationLoopbackPeer& client{ SerialCommunicationLoopback.Client() };
+};
```

Comment:
Can you hook up two SerialCommunicationMocks to the server and client, and test functionality via those mocks?

Reply:
Instead of, or as additional tests?